upgrading/extending-daml-models/1_package_ids.md

```diff
@@ -0,0 +1,104 @@
+You've already learned how to extend a DAML model when you added the messaging feature to the mini
+social network of the [Getting Started
+Guide](https://daml.com/learn/getting-started/your-first-feature). 
+
+There's a catch though with extending a DAML model like this. A [DAML
+archive](https://docs.daml.com/daml/reference/packages.html) is build with the `daml build` command.
+It contains the compiled artefact of your DAML code that will be deployed and run on the ledger.
+The package ID of a DAML archive depends on the contents of the compiled DAML code. Changing the
+code results in a different package ID. This means that contracts created with the old package will
+be different from contracts created with the updated code and if you would just upload the new
+package to your ledger and replace the old one, you couldn't execute any choices on contracts
+anymore, that have been created with the original code.
+
+While this behaviour presents a challenge to a ledger operator, it is a blessing for your data
+integrity. It gives you a guarantee that the read and write rights to your data can not be changed
+in hindsight, and stay exactly as you defined it.
+
+Let's verify this.
+
+First, open `daml/User.daml`{{open}} in the IDE and **wait for it to load**. This is the familiar
+`User` module of the Getting Started Guide.
```

Comment:
```suggestion
First, **open the IDE and wait for it to load**. After that open the `daml/User.daml`{{open}} file. This is the familiar `User` module of the Getting Started Guide.
```
Really a minor thing: user needs to open the IDE and wait for it to load and then open the file.